crates/holochain/src/conductor/conductor.rs

```diff
@@ -145,44 +144,103 @@ pub use state_impls::*;
 
 mod graft_records_onto_source_chain;
 
+/// How long we should attempt to achieve a "well-connected" CellStatus when first activating a cell,
+/// before moving on and letting the network health activity go on in the background.
+pub const JOIN_NETWORK_WAITING_PERIOD: std::time::Duration = std::time::Duration::from_secs(5);
+
+
 /// A list of Cells which failed to start, and why
 pub type CellStartupErrors = Vec<(CellId, CellError)>;
 
 /// Cloneable reference to a Conductor
 pub type ConductorHandle = Arc<Conductor>;
 
-/// The reason why a cell is waiting to join the network.
+/// The status of an installed Cell, which captures different phases of its lifecycle
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub enum PendingJoinReason {
-    /// The initial state, no attempt has been made to join the network yet.
-    Initial,
+pub enum CellStatus {
+    /// The cell is fully "online" and well-connected to a good number of peers
+    /// who know of its existence.
+    ///
+    /// This is the desired state for a cell, and should be maintained.
+    WellConnected,
+
+    /// The cell is connected to some peers, but there is evidence that there could
+    /// be more done to get to a better-connected state. Gossip may be weak and
+    /// network performance in general may be poor.
+    PartiallyConnected(String),
```

Comment:
I'm not convinced it is possible / useful to distinguish between these two cases. This really represents a gradient between I can only contact 1 out of an unknown number of peers to I can contact all known peers and they in turn at least don't know about any others, to everything in between.

Would any functionality actually behave different depending on which of these states we decide to set it to based on some mid-point threshold?

Reply:
Yeah I agree. I was trying to maintain some sense of what "joining" means. A successful "join" corresponds to WellConnected. But even a failed join may get you "partially connected".

If we distance ourselves even further from "join" and think "network health" instead, then we probably don't need these distinctions, and our ongoing attempts to improve and maintain health can be driven by actual metrics rather than some coarse-grained status.